cryptosystem/cryptosystem_utils.py

```diff
@@ -0,0 +1,80 @@
+# Utils used by cryptosystem modules.
+
+from random import getrandbits, randint
+
+
+# Generates two random primes.
+def generate_primes(bit_length):
+    prime_p = generate_random_prime(bit_length)
+    prime_q = generate_random_prime(bit_length)
+
+    while prime_q == prime_p:
+        prime_q = generate_random_prime(bit_length)
+
+    return prime_p, prime_q
+
+
+# Generates random prime with bit_length number of bits.
+def generate_random_prime(bit_length):
+    most_significant_bit = 1 << (bit_length-1)
+
+    p = getrandbits(bit_length-1)
+    while not is_prime(p | most_significant_bit):
+        p = getrandbits(bit_length-1)
+
+    return p | most_significant_bit
+
+
+# Checks whether or not given integer is a prime.
+# This probably should be a Miller-Rabin test if Python doesn't have a library.
+def is_prime(p):
```

Comment:
return p > 1 and all(p % i * i != 0 for i in range(int(p ** 0.5)))

Reply:
It's a stub and should be changed to more efficient function (it works forever for numbers > 10^18)
I'd WontFix and assign to @Omelianenko